coalib/results/Diff.py

```diff
@@ -278,18 +276,37 @@ def modified(self):
         return result
 
     @property
+    def modified(self):
+        """
+        Calculates the modified file, after applying the Diff to the original.
+        Also adds linebreaks at the end of each line (except eof)
+        """
+        eof_linebreak = False
+
+        if self._file and self._file[-1].endswith('\n'):
+            eof_linebreak = True
+
+        file = [line.rstrip('\n') for line in self._modified()]
+
+        if eof_linebreak:
+            return self._add_linebreaks(file)
+        return self._generate_linebreaks(file)
+
+    @property
     def unified_diff(self):
         """
         Generates a unified diff corresponding to this patch.
 
         Note that the unified diff is not deterministic and thus not suitable
         for equality comparison.
         """
-        return ''.join(difflib.unified_diff(
-                self.original,
-                self.modified,
+        list_unified_diff = list(difflib.unified_diff(
+                self._file,
+                self._modified(),
```

Comment:
Anyway, the `Diff.unified_diff` functions is used nowhere in the core except for the json output.
The printing in `ShowPatchAction` is done by using the difflib version directly. 

Interestingly, I don't see the issue you describe in #2006, even with the unmodified coala version.
Probably due to how printing the diff is handled in `ShowPatchAction.apply`, or rather in the printer, that adds a linebreak anyway ;)

In the end the question is, if the newline stuff in the diff is necessary, or for which usecase.